pkg/resources/resource_source_postgres_test.go

```diff
@@ -52,3 +56,41 @@ func TestResourceSourcePostgresCreate(t *testing.T) {
 		}
 	})
 }
+
+func TestResourceSourcePostgresUpdate(t *testing.T) {
+	r := require.New(t)
+	d := schema.TestResourceDataRaw(t, SourcePostgres().Schema, inSourcePostgres)
+
+	d.SetId("u1")
+	d.Set("size", "large")
+	r.NotNil(d)
+
+	testhelpers.WithMockDb(t, func(db *sqlx.DB, mock sqlmock.Sqlmock) {
+		mock.ExpectExec(`ALTER SOURCE "database"."schema"."source" SET \(SIZE = 'small'\)`).WillReturnResult(sqlmock.NewResult(1, 1))
+		mock.ExpectExec(`ALTER SOURCE "database"."schema"."" RENAME TO "source"`).WillReturnResult(sqlmock.NewResult(1, 1))
+		mock.ExpectExec(`ALTER SOURCE "database"."schema"."source" ADD SUBSOURCE "name1" AS "alias", "name2"`).WillReturnResult(sqlmock.NewResult(1, 1))
```

Comment:
Doesn't look like there's a test for `DROP SUBSOURCE`.

Reply:
Mentioned in more detail below, but the unit tests for updates are hard and kept to a minimum for the most part.